<?xml version="1.0" encoding="UTF-8" ?>
<launch>

    <include file="$(dirname)/perception/laplacian_line_detector_front.launch"/>

    <include file="$(dirname)/perception/lines_projector.launch">
        <arg name="use_camera_manager" value="true"/>
    </include>

    <include file="$(dirname)/perception/startlight_watcher.launch"/>
    <include file="$(dirname)/perception/finish_line_watcher.launch"/>

    <node pkg="rr_common" type="planner" name="planner" output="screen">
        <rosparam command="load" file="$(find rr_iarrc)/conf/planner_circuit.yaml"/>
    </node>

    <node name="navigation_controller" pkg="rr_common" type="navigation_controller" output="screen">
        <param name="req_finish_line_crosses" type="int" value="1"/>
        <param name="startSignal" type="string" value="/start_detected"/>
        <param name="resetSignal" type="string" value="/reset_detected"/>
        <param name="finishLineCrossesSignal" type="string" value="/finish_line_crosses"/>
        <param name="steeringAfterFinishTime" type="double" value="0.7"/>
    </node>

</launch>
